File: SDKs/Aspose.Cells_Cloud_SDK_For_CSharp/src/Com/Aspose/cells/Model/Style.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Cells.Model {
  public class Style {
    public Font Font { get; set; }

    public string Name { get; set; }

    public string CultureCustom { get; set; }

    public string Custom { get; set; }

    public Color BackgroundColor { get; set; }

    public Color ForegroundColor { get; set; }

    public bool? IsFormulaHidden { get; set; }

    public bool? IsDateTime { get; set; }

    public bool? IsTextWrapped { get; set; }

    public bool? IsGradient { get; set; }

    public bool? IsLocked { get; set; }

    public bool? IsPercent { get; set; }

    public bool? ShrinkToFit { get; set; }

    public int? IndentLevel { get; set; }

    public int? Number { get; set; }

    public int? RotationAngle { get; set; }

    public string Pattern { get; set; }

    public string TextDirection { get; set; }

    public string VerticalAlignment { get; set; }

    public string HorizontalAlignment { get; set; }

    public List<Border> BorderCollection { get; set; }

    public Link link { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Style {\n");
      sb.Append("  Font: ").Append(Font).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  CultureCustom: ").Append(CultureCustom).Append("\n");
      sb.Append("  Custom: ").Append(Custom).Append("\n");
      sb.Append("  BackgroundColor: ").Append(BackgroundColor).Append("\n");
      sb.Append("  ForegroundColor: ").Append(ForegroundColor).Append("\n");
      sb.Append("  IsFormulaHidden: ").Append(IsFormulaHidden).Append("\n");
      sb.Append("  IsDateTime: ").Append(IsDateTime).Append("\n");
      sb.Append("  IsTextWrapped: ").Append(IsTextWrapped).Append("\n");
      sb.Append("  IsGradient: ").Append(IsGradient).Append("\n");
      sb.Append("  IsLocked: ").Append(IsLocked).Append("\n");
      sb.Append("  IsPercent: ").Append(IsPercent).Append("\n");
      sb.Append("  ShrinkToFit: ").Append(ShrinkToFit).Append("\n");
      sb.Append("  IndentLevel: ").Append(IndentLevel).Append("\n");
      sb.Append("  Number: ").Append(Number).Append("\n");
      sb.Append("  RotationAngle: ").Append(RotationAngle).Append("\n");
      sb.Append("  Pattern: ").Append(Pattern).Append("\n");
      sb.Append("  TextDirection: ").Append(TextDirection).Append("\n");
      sb.Append("  VerticalAlignment: ").Append(VerticalAlignment).Append("\n");
      sb.Append("  HorizontalAlignment: ").Append(HorizontalAlignment).Append("\n");
      sb.Append("  BorderCollection: ").Append(BorderCollection).Append("\n");
      sb.Append("  link: ").Append(link).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }

File: SDKs/Aspose.Cells-Cloud-SDK-for-Ruby/lib/aspose_cells_cloud/models/style.rb
module AsposeCellsCloud
  # 
  class Style < BaseObject
    attr_accessor :font, :name, :culture_custom, :custom, :background_color, :foreground_color, :is_formula_hidden, :is_date_time, :is_text_wrapped, :is_gradient, :is_locked, :is_percent, :shrink_to_fit, :indent_level, :number, :rotation_angle, :pattern, :text_direction, :vertical_alignment, :horizontal_alignment, :border_collection, :link
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'font' => :'Font',
        
        # 
        :'name' => :'Name',
        
        # 
        :'culture_custom' => :'CultureCustom',
        
        # 
        :'custom' => :'Custom',
        
        # 
        :'background_color' => :'BackgroundColor',
        
        # 
        :'foreground_color' => :'ForegroundColor',
        
        # 
        :'is_formula_hidden' => :'IsFormulaHidden',
        
        # 
        :'is_date_time' => :'IsDateTime',
        
        # 
        :'is_text_wrapped' => :'IsTextWrapped',
        
        # 
        :'is_gradient' => :'IsGradient',
        
        # 
        :'is_locked' => :'IsLocked',
        
        # 
        :'is_percent' => :'IsPercent',
        
        # 
        :'shrink_to_fit' => :'ShrinkToFit',
        
        # 
        :'indent_level' => :'IndentLevel',
        
        # 
        :'number' => :'Number',
        
        # 
        :'rotation_angle' => :'RotationAngle',
        
        # 
        :'pattern' => :'Pattern',
        
        # 
        :'text_direction' => :'TextDirection',
        
        # 
        :'vertical_alignment' => :'VerticalAlignment',
        
        # 
        :'horizontal_alignment' => :'HorizontalAlignment',
        
        # 
        :'border_collection' => :'BorderCollection',
        
        # 
        :'link' => :'link'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'font' => :'Font',
        :'name' => :'String',
        :'culture_custom' => :'String',
        :'custom' => :'String',
        :'background_color' => :'Color',
        :'foreground_color' => :'Color',
        :'is_formula_hidden' => :'BOOLEAN',
        :'is_date_time' => :'BOOLEAN',
        :'is_text_wrapped' => :'BOOLEAN',
        :'is_gradient' => :'BOOLEAN',
        :'is_locked' => :'BOOLEAN',
        :'is_percent' => :'BOOLEAN',
        :'shrink_to_fit' => :'BOOLEAN',
        :'indent_level' => :'Integer',
        :'number' => :'Integer',
        :'rotation_angle' => :'Integer',
        :'pattern' => :'String',
        :'text_direction' => :'String',
        :'vertical_alignment' => :'String',
        :'horizontal_alignment' => :'String',
        :'border_collection' => :'Array<Border>',
        :'link' => :'Link'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'Font']
        self.font = attributes[:'Font']
      end
      
      if attributes[:'Name']
        self.name = attributes[:'Name']
      end
      
      if attributes[:'CultureCustom']
        self.culture_custom = attributes[:'CultureCustom']
      end
      
      if attributes[:'Custom']
        self.custom = attributes[:'Custom']
      end
      
      if attributes[:'BackgroundColor']
        self.background_color = attributes[:'BackgroundColor']
      end
      
      if attributes[:'ForegroundColor']
        self.foreground_color = attributes[:'ForegroundColor']
      end
      
      if attributes[:'IsFormulaHidden']
        self.is_formula_hidden = attributes[:'IsFormulaHidden']
      end
      
      if attributes[:'IsDateTime']
        self.is_date_time = attributes[:'IsDateTime']
      end
      
      if attributes[:'IsTextWrapped']
        self.is_text_wrapped = attributes[:'IsTextWrapped']
      end
      
      if attributes[:'IsGradient']
        self.is_gradient = attributes[:'IsGradient']
      end
      
      if attributes[:'IsLocked']
        self.is_locked = attributes[:'IsLocked']
      end
      
      if attributes[:'IsPercent']
        self.is_percent = attributes[:'IsPercent']
      end
      
      if attributes[:'ShrinkToFit']
        self.shrink_to_fit = attributes[:'ShrinkToFit']
      end
      
      if attributes[:'IndentLevel']
        self.indent_level = attributes[:'IndentLevel']
      end
      
      if attributes[:'Number']
        self.number = attributes[:'Number']
      end
      
      if attributes[:'RotationAngle']
        self.rotation_angle = attributes[:'RotationAngle']
      end
      
      if attributes[:'Pattern']
        self.pattern = attributes[:'Pattern']
      end
      
      if attributes[:'TextDirection']
        self.text_direction = attributes[:'TextDirection']
      end
      
      if attributes[:'VerticalAlignment']
        self.vertical_alignment = attributes[:'VerticalAlignment']
      end
      
      if attributes[:'HorizontalAlignment']
        self.horizontal_alignment = attributes[:'HorizontalAlignment']
      end
      
      if attributes[:'BorderCollection']
        if (value = attributes[:'BorderCollection']).is_a?(Array)
          self.border_collection = value
        end
      end
      
      if attributes[:'link']
        self.link = attributes[:'link']
      end
      
    end

  end
end
